### DISCUSSION TITLE
Custom headers not being sent along with the requests in createPagesServerClient (NextJS)

### DISCUSSION THREAD

**Discussion by Eerkz** [supabase/supabase#19481] [Feature Requests]
Here's how i'm passing the custom headers:

    const supabase = createPagesServerClient(
      { req, res },
      {
        supabaseUrl: "",
        supabaseKey: "",
        options: {
          global: {
            headers: {
              "x-custom-header": "value",
            },
          },
        },
      }
    );
	 // usage
	  const { data, error } = await supabase.storage
      .from("sample_bucket")
      .upload(`/${fileName}`, decodedImage, {
        upsert: true,
        contentType: dataType,
      });
I&apos;m trying to access this custom header in a function meant to run in an RLS policy for a supabase storage bucket. Here&apos;s the policy:

    ((bucket_id = 'sample_bucket'::text) AND some_function_to_run(((current_setting('request.headers::text, true))::json ->> 'x-custom-header'::text)))

The `some_function_to_run ` function always receives a null input.  I tried debugging things to see what does current_setting('request.heders') return by logging the it in the `some_function_to_run` function and it indeed doesn&apos;t include my custom header. Not really sure what i&apos;m doing wrong here. Here&apos;s the `some_function_to_run` implementation:

    CREATE OR REPLACE FUNCTION some_function_to_run(input_token TEXT)
    RETURNS BOOLEAN AS $$
    BEGIN
        RAISE LOG 'input_token: %', input_token;
        RETURN EXISTS (
            SELECT 1
            FROM session_tokens_table
            WHERE token = input_token
              AND NOT is_used
              AND expiration > NOW()
        );
    END;
    $$ LANGUAGE plpgsql SECURITY DEFINER;
    
I've also tried using createClient() instead but still having the same problem.  

I&apos;m using Next.js v13.4.10 and supabse-js v2.26.0.

**Comment by GaryAustin1** [comment]
Can you test that out on just a database postgrest rpc call?
I'm not sure storage is using postgrest so may not get the header.  I think they build up the jwt claims for auth.jwt() on their own.
But not 100%

**Comment by Eerkz** [comment]
> Can you test that out on just a database postgrest rpc call? I'm not sure storage is using postgrest so may not get the header. I think they build up the jwt claims for auth.jwt() on their own. But not 100%

I've tried what you suggested and createPagesServerClient custom headers does work with rpc calls.

**Comment by GaryAustin1** [comment]
So it is not a bug then.  It is just not currently supported in storage and would need to be a feature request.
